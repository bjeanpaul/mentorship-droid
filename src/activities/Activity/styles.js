import { StyleSheet } from 'react-native';


export default StyleSheet.create({
  poster: {
    width: 360,
    height: 248,
  },
});
