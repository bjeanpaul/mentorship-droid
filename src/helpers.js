import { find, isUndefined, partialRight as partial } from 'lodash';
import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const omitNulls = d => omitBy(d, isNull);


export const trap = (type, fn) => v => v instanceof type
  ? Promise.resolve(v).then(fn)
  : Promise.reject(v);


export const switchError = cases => (e, ...xargs) => {
  const [type, fn] = find(cases, ([type, fn]) => e instanceof type) || [];
  return !isUndefined(fn)
    ? Promise.resolve().then(() => fn(e, ...xargs))
    : Promise.reject(e);
};


export const apiAction = ({
  method,
  request,
  success,
  failures,
}) => (...args) => (dispatch, { auth }) => Promise.resolve()
  .then(() => request(...args))
  .then(dispatch)
  .then(() => method(...args, auth))
  .then(partial(success, ...args), partial(switchError(failures), ...args))
  .then(dispatch);


export const staticAction = type => () => ({
  type
});


export const dataAction = type => data => ({
  type,
  payload: data
});


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
