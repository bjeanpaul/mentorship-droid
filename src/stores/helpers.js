import moment from 'moment';
import { values, isUndefined, sortBy, filter, first } from 'lodash';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


// TODO sort?
export const getCategories = ({ entities: { categories } }) => values(categories);


// TODO sort?
export const getCategoryActivities = ({
  entities: { activities },
}, targetCategoryId) => (
  values(activities)
    .filter(({ category }) => category === targetCategoryId));


export const getCategory = ({ entities: { categories } }, id) => categories[id];


export const getActivity = ({ entities: { activities } }, id) => activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) => values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) => scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall && scheduledCall.activity && getActivity(state, scheduledCall.activity);
};


export const getEvents = ({ entities: { events } }) => values(events);


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) => (
  filter(callNotes, { callActivity: activityId }));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => callTime);
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};
