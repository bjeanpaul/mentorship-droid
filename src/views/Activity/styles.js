import { StyleSheet } from 'react-native';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import colors from 'src/constants/colors';


export default StyleSheet.create({
  categoryTitle: {
    marginLeft: 48,
    marginRight: 48,
  },
  posterContainer: {
    alignItems: 'center',
  },
  poster: {
    width: 360,
    height: 248,
    paddingTop: 1,
    paddingBottom: 1,
  },
  title: {
    fontSize: 32,
    fontFamily: FONT.MEDIUM,
    fontWeight: FONT_WEIGHT.MEDIUM,
    marginLeft: 24,
    marginRight: 24,
    paddingTop: 9,
    paddingBottom: 9,
  },
  about: {
    paddingTop: 10,
    paddingBottom: 24,
  },
  shortcuts: {
    paddingTop: 23,
    paddingBottom: 24,
  },
  status: {
    color: colors.ACTIVITY_STATUS,
    fontSize: 14,
    marginLeft: 24,
    marginRight: 24,
  },
  statusAbout: {
    paddingBottom: 27,
  },
  statusShortcuts: {
    paddingBottom: 19,
  },
  actionContainer: {
    marginLeft: 24,
    marginRight: 24,
    paddingTop: 15,
    paddingBottom: 17,
    borderColor: colors.TEXT_LIGHT,
    borderWidth: 2,
    borderRadius: 4,
  },
  actionContainerShortcuts: {
    marginTop: 0,
  },
  actionText: {
    fontSize: 13,
    letterSpacing: 1,
    color: colors.TEXT_LIGHT,
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
  },
  section: {
    marginBottom: 29,
  },
  sectionHeader: {
    paddingLeft: 24,
    paddingRight: 24,
    height: 72,
    backgroundColor: colors.ACTIVITY_SECTION_HEADER_BG,
    borderColor: colors.ACTIVITY_SECTION_HEADER_BORDER,
    justifyContent: 'center',
    marginBottom: 12,
  },
  sectionIcon: {
    width: 40,
    height: 40,
    top: 16,
    position: 'absolute',
    resizeMode: 'contain',
  },
  sectionTitle: {
    fontSize: 12,
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
    color: colors.ACTIVITY_SECTION_HEADER_TITLE,
  },
  sectionBody: {
    paddingLeft: 24,
    paddingRight: 24,
  },
});
