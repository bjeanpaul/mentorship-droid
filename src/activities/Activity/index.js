import React, { PropTypes } from 'react';
import { Image } from 'react-native';

import { BaseView, Header, HeaderIcon, Text } from 'src/components';
import styles from './styles';


const Activity = ({
  category: { color },
  activity: {
    title,
    poster,
  },
  onBackPress,
}) => (
  <BaseView>
    <Header style={{ backgroundColor: color }}>
      <HeaderIcon
        uid="back"
        type={HeaderIcon.types.backLight}
        onPress={onBackPress}
      />

      <Text style={[Text.types.title, Text.themes.light]}>{title}</Text>
    </Header>

    {
      poster
        ? <Image
          source={{ uri: poster }}
          style={[styles.poster, { backgroundColor: color }]}
        />
        : null
    }
  </BaseView>
);


Activity.propTypes = {
  category: PropTypes.object.isRequired,
  activity: PropTypes.object.isRequired,
  onBackPress: PropTypes.func.isRequired,
};


export default Activity;
