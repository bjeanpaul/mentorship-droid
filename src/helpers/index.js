import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const conj = (...args) => Object.assign({}, ...args);


export const omitNulls = d => omitBy(d, isNull);


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
