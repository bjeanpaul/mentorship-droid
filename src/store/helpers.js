import moment from 'moment';
import { values, isUndefined, sortBy, filter, first, orderBy } from 'lodash';

import { getCurrent } from 'src/navigationHelpers';
import { TYPES_TO_COLLECTIONS } from 'src/constants/events';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


export const getCategories = ({ entities }) =>
  sortBy(values(entities.categories), 'ordinal');


export const getCategoryActivities = ({ entities }, targetCategoryId) => {
  const activities = values(entities.activities)
    .filter(({ category }) => category === targetCategoryId);

  return sortBy(activities, 'ordinal');
};


export const getCategory = ({ entities: { categories } }, id) =>
  categories[id];


export const getActivity = ({ entities: { activities } }, id) =>
  activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) =>
  values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) =>
  scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall
      && scheduledCall.activity
      && getActivity(state, scheduledCall.activity);
};


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) =>
  values(callNotes)
    .filter(({ callActivityId }) => callActivityId === activityId);


export const getCallNotes = ({
  entities: { callNotes },
}) =>
  sortBy(callNotes, ({ callStartTime }) => +moment(callStartTime));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => +moment(callTime));
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};


export const getScheduledCallsBetween = (state, start, end, time = Date.now()) => {
  // for some reason, moment durations are behaving inconsistently in tests,
  // thus the lack of moment usage here
  const startTime = +moment(time) - start;
  const endTime = +moment(time) + end;

  return getScheduledCalls(state)
    .filter(({ callTime }) => {
      const t = +moment(callTime);
      return startTime <= t && t <= endTime;
    });
};


const getEventObject = (state, event) => {
  const collection = state.entities[TYPES_TO_COLLECTIONS[event.eventType]];
  return collection && collection[event.objectId];
};


// TODO return the associated entitites to make the containers' jobs easier
export const getEvents = (state) =>
  values(state.entities.events)
    .filter(event => getEventObject(state, event));


export const getMessages = ({ entities }) => []
  .concat(values(entities.messages))
  .concat(values(entities.pendingMessages));


export const getMessage = ({ entities }, id) => entities.messages[id];


export const getActiveTopRoute = store => getCurrent(store.navigation.top);


export const getBlogPost = ({ entities }, id) => entities.blogPosts[id];


export const getBlogPosts = ({ entities }) =>
  orderBy(entities.blogPosts, [({ createdAt }) => +moment(createdAt)], ['desc']);
