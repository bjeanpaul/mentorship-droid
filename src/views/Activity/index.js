import moment from 'moment';
import React, { PropTypes } from 'react';
import { View, Image, TouchableNativeFeedback, ScrollView } from 'react-native';

import { BaseView, Header, HeaderIcon, Text, Panel } from 'src/components';
import images from 'src/constants/images';
import styles from './styles';


const Activity = props => {
  const {
    category: {
      color,
      title: categoryTitle,
    },
    activity: {
      id,
      title,
      poster,
      objective,
      lessonRationale,
      instructions,
      prompts,
      reflectionPoints,
      skillsDeveloped,
    },
    onBackPress,
  } = props;

  return (
    <BaseView key={`activity:${id}`}>
      <ScrollView>
        <Header style={{ backgroundColor: color }}>
          <Text
            numberOfLines={1}
            style={[Text.types.title, Text.themes.light, styles.categoryTitle]}
          >
            {categoryTitle}
          </Text>

          <HeaderIcon
            uid="back"
            type={HeaderIcon.types.backLight}
            onPress={onBackPress}
          />
        </Header>

        {
          poster.exists() && <View style={[styles.posterContainer, { backgroundColor: color }]}>
            <Image
              source={poster.resize(360, 248).toSource()}
              style={styles.poster}
            />
          </View>
        }

        <View style={[styles.about, { backgroundColor: color }]}>
          <Text
            numberOfLines={1}
            style={[Text.themes.light, styles.title]}
          >
            {title}
          </Text>

          <Status style={styles.statusAbout} {...props} />
          <Action uid="topAction" {...props} />
        </View>

        <Section
          icon={images.ACTIVITY_OBJECTIVE}
          title="Objective"
        >
          {objective}
        </Section>

        <Section
          icon={images.ACTIVITY_RATIONALE}
          title="Lesson Rationale"
        >
          {lessonRationale}
        </Section>

        <Section
          icon={images.ACTIVITY_INSTRUCTIONS}
          title="Instructions"
        >
          {instructions}
        </Section>

        <Section
          icon={images.ACTIVITY_PROMPTS}
          title="Tips"
        >
          {prompts}
        </Section>

        <Section
          icon={images.ACTIVITY_REFLECTION_POINTS}
          title="Reflection Points"
        >
          {reflectionPoints}
        </Section>

        <Section
          icon={images.ACTIVITY_SKILLS_DEVELOPED}
          title="Skills Developed"
        >
          {skillsDeveloped}
        </Section>

        <View style={[styles.shortcuts, { backgroundColor: color }]}>
          <Status style={styles.statusShortcuts} {...props} />
          <Action
            uid="bottomAction"
            style={{ container: styles.actionContainerShortcuts }} {...props}
          />
        </View>
      </ScrollView>
    </BaseView>
  );
};


const Section = ({
  children,
  ...props,
}) => (
  <Panel {...props}>
    <Text style={Text.types.paragraph}>
      {children}
    </Text>
  </Panel>
);

const Status = ({
  style,
  latestCall,
  nextScheduledCall,
}) => {
  const statusStyles = [styles.status, style];

  if (latestCall) {
    return (
      <Text style={statusStyles}>
        Last discussed on {moment(latestCall.startTime).format('MMM D, h:mm a')}
      </Text>
    );
  } else if (nextScheduledCall) {
    return (
      <Text style={statusStyles}>
        Scheduled to discuss on {moment(nextScheduledCall.callTime).format('MMM D, h:mm a')}
      </Text>
    );
  } else {
    return (
      <Text style={statusStyles}>
        Discuss this activity with your Mentee
      </Text>
    );
  }
};


const Action = ({
  style,
  activity: { id },
  latestCall,
  nextScheduledCall,
  onSchedulePress,
  onReschedulePress,
  onViewCallNotesPress,
}) => {
  if (latestCall) {
    return (
      <ActionButton style={style} onPress={() => onViewCallNotesPress(id)}>
        Add or view notes
      </ActionButton>
    );
  } else if (nextScheduledCall) {
    return (
      <ActionButton style={style} onPress={() => onReschedulePress(nextScheduledCall.id)}>
        Rechedule Call
      </ActionButton>
    );
  } else {
    return (
      <ActionButton style={style} onPress={() => onSchedulePress(id)}>
        Schedule Call
      </ActionButton>
    );
  }
};


const ActionButton = ({
  onPress,
  children,
  style = {},
}) => (
  <TouchableNativeFeedback onPress={onPress}>
    <View style={[styles.actionContainer, style.container]}>
      <Text style={[Text.uppercase, styles.actionText, style.action]}>
        {children}
      </Text>
    </View>
  </TouchableNativeFeedback>
);


Activity.propTypes = {
  category: PropTypes.object.isRequired,
  activity: PropTypes.object.isRequired,
  onBackPress: PropTypes.func.isRequired,
};


Section.propTypes = {
  children: PropTypes.any,
};


Status.propTypes = {
  style: PropTypes.any,
  latestCall: PropTypes.object,
  nextScheduledCall: PropTypes.object,
};


Action.propTypes = {
  style: PropTypes.any,
  activity: PropTypes.object.isRequired,
  latestCall: PropTypes.object,
  nextScheduledCall: PropTypes.object,
  onSchedulePress: PropTypes.func.isRequired,
  onReschedulePress: PropTypes.func.isRequired,
  onViewCallNotesPress: PropTypes.func.isRequired,
};


ActionButton.propTypes = {
  style: PropTypes.object,
  onPress: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};


export default Activity;
