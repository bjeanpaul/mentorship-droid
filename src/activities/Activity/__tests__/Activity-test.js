import { noop } from 'lodash';
import React from 'react';

import Activity from 'src/activities/Activity';
import { fakeCategory, fakeActivity, uidEquals } from 'app/scripts/helpers';
import { ACTIVITY_POSTER } from 'app/scripts/fixtures';


describe('Activity', () => {
  const createComponent = (props = {}) => (
    <Activity
      category={fakeCategory()}
      activity={fakeActivity({ title: 'Activity 1' })}
      onBackPress={noop}
      onSchedulePress={noop}
      {...props}
    />
  );

  it('should render', () => {
    const el = render(createComponent());
    expect(el).toMatchSnapshot();
  });

  it('should not try display a poster if none is given', () => {
    const el = render(createComponent({ poster: null }));
    expect(el).toMatchSnapshot();
  });

  it('should display the poster if one is given', () => {
    const el = render(createComponent({ poster: ACTIVITY_POSTER }));
    expect(el).toMatchSnapshot();
  });

  it('should call onBackPress when the back button is pressed', () => {
    const onBackPress = jest.fn();
    const el = shallow(createComponent({ onBackPress }));

    el.findWhere(uidEquals('back'))
      .simulate('press');

    expect(onBackPress.mock.calls).toEqual([[]]);
  });

  fit('should call onSchedulePress when the top schedule button is pressed', () => {
    const onSchedulePress = jest.fn();

    shallow(createComponent({ onSchedulePress }))
      .find('Action')
      .shallow()
      .find('ActionButton')
      .shallow()
      .simulate('press');

    expect(onSchedulePress.mock.calls).toEqual([[]]);
  });
});
