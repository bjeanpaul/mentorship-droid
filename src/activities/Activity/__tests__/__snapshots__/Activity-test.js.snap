exports[`Activity should display the poster if one is given 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16
        },
        Object {
          "backgroundColor": "#97c13c"
        }
      ]
    }>
    <View
      onPress={[Function noop]}
      style={
        Array [
          Object {
            "position": "absolute",
            "top": 24
          },
          Object {}
        ]
      }>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 24,
              "width": 24
            },
            Object {}
          ]
        } />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400"
            },
            Object {
              "color": "#fff"
            }
          ]
        ]
      }>
      Activity 1
    </Text>
  </View>
</View>
`;

exports[`Activity should not try display a poster if none is given 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16
        },
        Object {
          "backgroundColor": "#97c13c"
        }
      ]
    }>
    <View
      onPress={[Function noop]}
      style={
        Array [
          Object {
            "position": "absolute",
            "top": 24
          },
          Object {}
        ]
      }>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 24,
              "width": 24
            },
            Object {}
          ]
        } />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400"
            },
            Object {
              "color": "#fff"
            }
          ]
        ]
      }>
      Activity 1
    </Text>
  </View>
</View>
`;

exports[`Activity should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16
        },
        Object {
          "backgroundColor": "#97c13c"
        }
      ]
    }>
    <View
      onPress={[Function noop]}
      style={
        Array [
          Object {
            "position": "absolute",
            "top": 24
          },
          Object {}
        ]
      }>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 24,
              "width": 24
            },
            Object {}
          ]
        } />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400"
            },
            Object {
              "color": "#fff"
            }
          ]
        ]
      }>
      Activity 1
    </Text>
  </View>
</View>
`;
